egs2/voices/asr1/local/data.sh

```diff
@@ -0,0 +1,212 @@
+        fi
+        mkdir -p data/${train_set}${suffix}_tmp data/${train_set}${suffix} data/${valid_set}${suffix} data/${test_set}${suffix}
+        # 12800 samples (devkit) or 661248 samples (full)
+        find "${root}"/${cond}/train -iname '*.wav' | \
+            awk -F '/' '{a=$NF; n=split(a, ret, ".wav"); print(ret[1]" "$0)}' | sort \
+            > data/${train_set}${suffix}_tmp/wav.scp
+        # 6400 samples (devkit) or 337920 samples (full)
+        find "${root}"/${cond}/test -iname '*.wav' | \
+            awk -F '/' '{a=$NF; n=split(a, ret, ".wav"); print(ret[1]" "$0)}' | sort \
+            > data/${test_set}${suffix}/wav.scp
+
+        if [ $cond = "distant-16k/speech" ]; then
+            # e.g., Lab41-SRI-VOiCES-rm1-babb-sp0198-ch000209-sg0010-mc01-stu-clo-dg030
+            awk '{print $1}' data/${train_set}${suffix}_tmp/wav.scp | \
+                awk -F '-' '{if (substr($6,1,2)!="sp") {exit 1} sid=substr($6,3,length($6)); print($0" "sid)}' \
+                > data/${train_set}${suffix}_tmp/utt2spk
+            awk '{print $1}' data/${test_set}${suffix}/wav.scp | \
+                awk -F '-' '{if (substr($6,1,2)!="sp") {exit 1} sid=substr($6,3,length($6)); print($0" "sid)}' \
+                > data/${test_set}${suffix}/utt2spk
+        else
+            # e.g., Lab41-SRI-VOiCES-src-sp0032-ch021631-sg0005
+            awk '{print $1}' data/${train_set}${suffix}_tmp/wav.scp | \
+                awk -F '-' '{if (substr($5,1,2)!="sp") {exit 1} sid=substr($5,3,length($5)); print($0" "sid)}' \
+                > data/${train_set}${suffix}_tmp/utt2spk
+            awk '{print $1}' data/${test_set}${suffix}/wav.scp | \
+                awk -F '-' '{if (substr($5,1,2)!="sp") {exit 1} sid=substr($5,3,length($5)); print($0" "sid)}' \
+                > data/${test_set}${suffix}/utt2spk
+        fi
+        for x in "${train_set}${suffix}_tmp" "${test_set}${suffix}"; do
+            awk 'FNR==NR {spk[$1]=$2; next} {print(spk[$1]"_"$0)}' \
+                data/${x}/utt2spk data/${x}/wav.scp | sort > data/${x}/wav_new.scp
+            mv data/${x}/wav_new.scp data/${x}/wav.scp
+            awk '{print($2"_"$0)}' data/${x}/utt2spk | sort > data/${x}/utt2spk_new
+            mv data/${x}/utt2spk_new data/${x}/utt2spk
+        done
+
+        utils/utt2spk_to_spk2utt.pl data/${train_set}${suffix}_tmp/utt2spk > data/${train_set}${suffix}_tmp/spk2utt
+        utils/utt2spk_to_spk2utt.pl data/${test_set}${suffix}/utt2spk > data/${test_set}${suffix}/spk2utt
+
+        utils/filter_scp.pl data/${train_set}${suffix}_tmp/wav.scp data/local/voices_text${suffix} | sort > data/${train_set}${suffix}_tmp/text
+        utils/filter_scp.pl data/${test_set}${suffix}/wav.scp data/local/voices_text${suffix} | sort > data/${test_set}${suffix}/text
+
+        # split ${train_set}${suffix}_tmp set to ${train_set} and ${valid_set}
+        head -n 10 data/${train_set}${suffix}_tmp/spk2utt > data/${train_set}${suffix}_tmp/dev_spk2utt
```

Comment:
Note that I always take utterances from the first 10 speakers as the dev set. I'm not sure whether this is a reasonable setting.
There is no official split for training and development sets.

Reply:
This results in the following amount of samples in each subset:
```bash
egs2/voices/asr1/data/
├── full_dev
│   ├── spk2utt (10 speakers)
│   ├── text
│   ├── utt2spk
│   └── wav.scp (33924 utterances)
├── full_dev_multich
│   ├── spk2utt (10 speakers)
│   ├── text
│   ├── utt2spk
│   └── wav.scp (6336 utterances)
├── full_test
│   ├── spk2utt (100 speakers)
│   ├── text
│   ├── utt2spk
│   └── wav.scp (339240 utterances)
├── full_test_multich
│   ├── spk2utt (100 speakers)
│   ├── text
│   ├── utt2spk
│   └── wav.scp (63360 utterances)
├── full_train
│   ├── spk2utt (190 speakers)
│   ├── text
│   ├── utt2spk
│   └── wav.scp (629907 utterances)
└── full_train_multich
    ├── spk2utt (190 speakers)
    ├── text
    ├── utt2spk
    └── wav.scp (117648 utterances)
```